### ISSUE TITLE
Release mode means no asserts

### ISSUE THREAD

**Issue by janisozaur** [mozilla/rr#1835]
Hi,
I just noticed in (release notes for 4.4.0)[https://github.com/mozilla/rr/releases/tag/4.4.0] this bit:

> Compile releases with `CMAKE_BUILD_TYPE=Release` for improved performance.

I just wanted to check you are aware it means `-O3 -D_NDEBUG` which will disable all asserts and perhaps other checks in the code, which in the past have been reported and fixed. CMake also has `RelWithDebInfo` build type, which is `-O2 -g -DDEBUG` and may perhaps be better suited.

**Comment by rocallahan** [comment]
> I just wanted to check you are aware it means -O3 -D_NDEBUG which will disable all asserts and perhaps other checks in the code, which in the past have been reported and fixed.

Yes. It doesn't disable rr's `ASSERT` and `FATAL` checks, which are pretty common.

> CMake also has RelWithDebInfo build type, which is -O2 -g -DDEBUG and may perhaps be better suited.

Hmm, I'll try that.

**Comment by rocallahan** [comment]
Our `CMakeList.txt` already has `-g3` for `Release` builds.

I don't really want to ship releases with `-DDEBUG` in the long term. We already have some nontrivially expensive checks behind `DEBUG` that should not, IMHO, run in release builds. I think

**Comment by janisozaur** [comment]
Is there a reason for this to stay open then?

**Comment by rocallahan** [comment]
It can stay open for the work described in my previous comment.
